### PR TITLE
[JENKINS-29264] Unsupported 'ps' output detected, warning generated t…

### PR DESCRIPTION
**Pull request by pjanouse**
…o server.log and build as well

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/proccleaner/PsKiller.java [hunk 1]
```diff
@@ -37,15 +38,24 @@
...
+        }
     }
 
     public void kill(String user, PrintStream log) throws InterruptedException, IOException {
         PsBasedProcessTree ptree = PsBasedProcessTreeFactory.createPsBasedProcessTree().createProcessTreeFor(user);
-        System.out.println("Process tree: " + ptree.toString());
-        ptree.setLog(log);
-        int me = ProcCleaner.getpid();
-        doKill(ptree, me);
+        if(ptree != null) {
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/proccleaner/PsKiller.java]: add space after if

File: src/main/java/org/jenkinsci/plugins/proccleaner/PsKiller.java [hunk 2]
```diff
@@ -57,4 +67,6 @@ public void kill(String user, PrintStream log) throws InterruptedException, IOEx
     }
 
     private static final long serialVersionUID = 1L;
+
+    private static final Logger logger = Logger.getLogger(PsBasedUnixProcessTree.class.getName());
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/proccleaner/PsKiller.java]: call variable uppercase LOGGER because it static.

**Comment by olivergondza** [comment]
One of the loggers introduced is unused. To me it seems it make more sense to log unsupported system in `PsBasedUnixProcessTree` instead of in (all) its clients.

**Comment by pjanouse** [comment]
Oliver,
un-used logger is fixed.

On the other hands, I hadn't wanted to change the current implementation so much, so I've relied on the current interfaces... The class PsBasedProcessTree isn't a client (caller) of the function createProcessTreeFor() (in addition it is  declared as an abstract method here, so I can't insert any code). The (only one) client for createProcessTreeFor() is the class PsKiller where I put the check.
I can remove the check code from PsKiller, but in that case I've to insert it to all inherited classes from PsBasedProcessTree which I would avoid.

Do you have other idea how to implement that without so much code duplication?

## 

Pavel Janousek
Senior JBoss QA Engineer

**Comment by vjuranek** [comment]
Merged, thanks!

**Comment by vjuranek** [comment]
Btw, couple of notes:
- you can add `[FIXED JENKINS-29264]` into the commit message, it will automatically resolve the JIRA once merged and also add a link to appropriate commit, so the reader of the issue can navigate to the commit very easily later on
- if you do some fixes to the original PR, it's not a bad idea to squash the commits, so that the peolpe who do the review and look on it commit by commit don't have to think about changes which are fixed in subsequent commits

**Pull request closed without merging**